src/Cli/dotnet/commands/InstallingWorkloadCommand.cs

```diff
@@ -133,7 +133,16 @@ public bool TryInstallWorkloadSet(ITransactionContext context, out IEnumerable<M
             if (File.Exists(Path.Combine(advertisingPackagePath, Constants.workloadSetVersionFileName)))
             {
                 // This file isn't created in tests.
-                PrintWorkloadSetTransition(File.ReadAllText(Path.Combine(advertisingPackagePath, Constants.workloadSetVersionFileName)));
+                var version = File.ReadAllText(Path.Combine(advertisingPackagePath, Constants.workloadSetVersionFileName));
+                PrintWorkloadSetTransition(version);
+                if (!version.Equals(_workloadSetVersion) && !version.Equals(_workloadSetVersionFromGlobalJson))
```

Comment:
Great catch!

Reply:
I think the best thing to do would be to update `WorkloadManifestUpdater.UpdateManifestWithVersionAsync` so that when it writes the version file and the `packageVersion` is specified, it checks to see whether they match and throws an exception if not.  Then we can remove the logic in `InstallingWorkloadCommand` to handle the difference.

If we ever end up in a situation where that exception is thrown, then we can investigate why and figure out what to do from there (which might include failing more gracefully if it's actually a valid behavior).

Reply:
Sounds good to me